### PR TITLE
HACKTOBERFEST - Added solution to Euler 64.

### PR DESCRIPTION
**Pull request by kakashi215**
This PR is for Hacktoberfest. #Hacktoberfest
Added Python solution to [Project Euler Problem 64.](https://projecteuler.net/problem=64)

### **Describe your change:**
Added my solution to the [Project Euler Problem 64.](https://projecteuler.net/problem=64)
Added a folder problem_064.
Added __init__.py file.
Added sol1.py file.
Fixes: #3705 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by kakashi215** [comment]
Requesting @dhruvmanila and @Kush1101 to review the commit.

Cheers!

**Comment by kakashi215** [comment]
@dhruvmanila @Kush1101 Anyone around?
Please could you review the PR?

**Comment by kakashi215** [comment]
@cclauss 
@dhruvmanila 
@Kush1101 

Guys please do the needful!

**Comment by xcodz-dot** [comment]
looks good, sorry for late response (You know the stats, right.)

@cclauss @dhruvmanila @Kush1101 @poyea @ruppysuppy, Possibly anyone check this out

**Comment by kakashi215** [comment]
> looks good, sorry for late response (You know the stats, right.)
> 
> @cclauss @dhruvmanila @Kush1101 @poyea @ruppysuppy, Possibly anyone check this out

Yeah man! You  guys have been flooded with PRs.
I'd love to help up any way posssible!!!!

File: project_euler/problem_064/sol1.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    count_odd_periods = 0
+    for i in range(2, n+1):
+        sr = sqrt(i)
+        if sr - floor(sr) != 0:
+            if continuous_fraction_period(i) % 2 == 1:
+                count_odd_periods += 1
+    return count_odd_periods
+
+if __name__ == "__main__":
+    print(f"{solution(int(input().split()))}")
```
Comment (ruppysuppy) [inline project_euler/problem_064/sol1.py:77]:
```Suggestion
if __name__ == "__main__":
    print(solution())
```

Replace `an` with a descriptive variable name

**Review by ruppysuppy — CHANGES_REQUESTED** [review]

File: project_euler/problem_064/sol1.py (hunk 1 above)
Comment (cclauss) [reply project_euler/problem_064/sol1.py:77]: The program seems much more valuable if it is not hardcoded and continues to enable experimentation.  Why remove the end user's ability to experiment with different input values?

Comment (kakashi215) [reply project_euler/problem_064/sol1.py:77]: On it with the changes!!

**Review by cclauss — APPROVED** [review]

File: project_euler/problem_064/sol1.py (hunk 1 above)
Comment (kakashi215) [reply project_euler/problem_064/sol1.py:77]: > The program seems much more valuable if it is not hardcoded and continues to enable experimentation. Why remove the end user's ability to experiment with different input values?

I think this allows you to call the solution function both with and without a value. When called without value, it accepts the 10000 default value given in the function declaration.

Comment (cclauss) [reply project_euler/problem_064/sol1.py:77]: @kakashi215 Yes.  I did read the code.  I like `print(f"{solution(int(input().split()))}")` the way that it is now and do not see how `print(f"solution() = {solution()}")` is an improvement.

Comment (kakashi215) [reply project_euler/problem_064/sol1.py:77]: @ruppysuppy i think the below code does the same and is preferred as the pythonic way.
print(f"{solution()=}")
So am changing it to that.
Lemme know if we need to change it.

**Comment by cclauss** [comment]
Please revert the last change.  It removes the end user's ability to experiment with different inputs.

File: project_euler/problem_064/sol1.py (hunk 1 above)
Comment (kakashi215) [reply project_euler/problem_064/sol1.py]: @cclauss 
This `print(f"{solution(int(input().split()))}")` would always need a value to be entered.
But this `print(f"solution() = {solution()}")` can work without any input as well. Although i agree that you'd have to enter custom value in code.

**Comment by kakashi215** [comment]
> Please revert the last change. It removes the end user's ability to experiment with different inputs.

On it.

File: project_euler/problem_064/sol1.py (hunk 1 above)
Comment (kakashi215) [reply project_euler/problem_064/sol1.py]: @ruppysuppy Based on the comments am resolving the issue.

**Review by ruppysuppy — APPROVED** [review]

**Pull request merged**